src/main/kotlin/com/example/ajaxproject/service/UserServiceImpl.kt

```diff
@@ -4,36 +4,43 @@ import com.example.ajaxproject.dto.UserDTO
 import com.example.ajaxproject.mapper.UserMapper
 import com.example.ajaxproject.model.User
 import com.example.ajaxproject.repository.UserRepository
+import com.example.ajaxproject.service.interfaces.UserService
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Service
 
 @Service
 class UserServiceImpl @Autowired constructor(
     private val userRepository: UserRepository,
     private val userMapper: UserMapper
-) {
-    fun findAll(): List<User> {
-        return userRepository.findAll()
-    }
+) : UserService {
 
-    fun findById(id: Long): User? {
-        return userRepository.findById(id).orElse(null)
-    }
-
-    fun createUser(userDTO: UserDTO): User {
+    override fun createUser(userDTO: UserDTO): User {
         val user = userMapper.toEntity(userDTO)
         return userRepository.save(user)
     }
 
-    fun updateUser(id: Long, userDTO: UserDTO): User {
-        val existingUser = userRepository.findById(id).orElseThrow { NoSuchElementException("User not found") }
-        val updatedUser = userMapper.toEntity(userDTO)
-        updatedUser.id = existingUser.id
-        return userRepository.save(updatedUser)
+    override fun updateUser(id: Long, userDTO: UserDTO): User {
+        val user = userRepository.findById(id)
+            .orElseThrow { NoSuchElementException("User not found") }
+
+        user.apply {
+            email = userDTO.email
+            password = userDTO.password
+        }
+
+        return userRepository.save(user)
     }
 
-    fun deleteUser(id: Long) {
+    override fun deleteUser(id: Long) {
         userRepository.deleteById(id)
     }
 
+    override fun findUserById(id: Long): User? {
+        return userRepository.findById(id).orElse(null)
```

Comment:
When you find entity by id and it was not found then throw exception like you did in `#updateUser`

Reply:
one more thing. if a method throws an exception on not found, it's better to name it like 'get...'

Reply:
Renamed all find methods to get